### DISCUSSION TITLE
picamera2 v picamera

### DISCUSSION THREAD

**Discussion by ktorimaru** [raspberrypi/picamera2#146] [Q&A] [answered]
This may seem like a stupid question.
Is picamera2 patterned after picamera?
I have never used picamera and was wondering if the recipies on [https://picamera.readthedocs.io/en/release-1.13/index.html](https://picamera.readthedocs.io/en/release-1.13/index.html) would be helpful in learning picamera2?

**Comment by davidplowman** [comment, marked as answer]
I wouldn't recommend learning the original Picamera as a way of finding out about Picamera2. Picamera2 follows the API of the open source libcamera project quite closely, which in turn exposes the features of the Pi's camera system fairly directly. The old Picamera provides access to a deprecated and proprietary Broadcom camera API.

We have started work on the official Picamera2 docuemtnation, however, it's going to take a while before it's very useful. In the meantime I'd encourage people to look at the project's [README file](https://github.com/raspberrypi/picamera2/blob/main/README.md), study the quite large number of [examples](https://github.com/raspberrypi/picamera2/tree/main/examples), and if those don't help, to raise an [issue](https://github.com/raspberrypi/picamera2/issues) for which there is a "HOW TO" sub-category. Hope that helps!

**Reply by ktorimaru** [reply]
Thank you
